pkg/vault/test/foundry/VaultSwap.t.sol

```diff
@@ -234,6 +234,36 @@ contract VaultSwapTest is BaseVaultTest {
         );
     }
 
+    function testSwapEventExactInRecovery() public {
```

Comment:
This also happens for add / remove :/

Should we add a small test too?

Reply:
That is harder. I tweaked some tests in VaultLiquidity.t.sol to set fees and unbalance, and verified in the logs that it was reporting the fee in PoolBalanceChanged, but it's hard to predict the values (rounding, etc.).

I think I'm out of time for that tonight; want to finish the rest of it.